Week3/homework/step4-bonus.js

```diff
@@ -1,10 +1,21 @@
 'use strict';
 
-const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
+// ----------first way to do it without es6-----------
 
-// Add your function here. Try and come up with a good name for this function
+// const different = (val, ind, self) => {
+//     return self.indexOf(val) === ind;
+// }
 
-// Replace `yourFunction` with the name of the function you just created
-const uniqueValues = yourFunction(values);
+// const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
+// values.filter(different);
+// // Add your function here. Try and come up with a good name for this function
+
+// console.log(values.filter(different));
+
+// ----------second way to do it with es6-------------
 
-console.log(uniqueValues);
+const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
+
+const newValues = [... new Set(values)]
```

Comment:
Nice use of the spread operator. 👍 Would you also like to try this out using `Array.reduce`?

Reply:
@yash-kapila How could I use reduce in this case???

Reply:
A good way to find a use-case for `Array.reduce` is that it can be used when we want to get a single value from an array and that value can be anything, a string, a number or an array itself.

So reduce accepts an accumulator which in our case is going to be our new array with no repetitions starting with an empty array `[]`. Then we iterate through each of elements in original array and then check if it exists inside our accumulator. If it does,  we don't do anything but if it doesn't, we simply push the new element to our accumulator. Something like this:

```
const uniqueArray = arr => arr.reduce((acc,elem) => {
  if (!acc.includes(elem)) {
    acc.push(elem);
  }
  return acc;
}, []);
```

I would agree though that using a Set is much simpler for this particular operation but with this way, you can perhaps improve your knowledge of `Array.reduce`. 🙂